### PR TITLE
Reduce parallelization to prevent model build from crashing.

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

**Review by smcclure17 — APPROVED** [review]

File: libs/parallel_utils.py [hunk 1]
```diff
@@ -33,7 +33,10 @@ def parallel_map(func: Callable[[T], R], iterable: Iterable[T]) -> Iterable[R]:
     if USE_MULTIPROCESSING:
         # Setting maxtasksperchild to one ensures that we minimize memory usage over time by creating
         # a new child for every task. Addresses OOMs we saw on highly parallel build machine.
-        with multiprocessing.Pool(maxtasksperchild=1) as pool:
+        # But that might not be enough. Also make sure we don't spawn more than 64 processes (the
+        # build machine is 96-core)
+        processes = min(os.cpu_count(), 64)
```
Comment (smcclure17) [inline libs/parallel_utils.py:38]: just for curiosity, is there a specific reason for choosing 64 or is it arbitrary?

Comment (mikelehen) [reply libs/parallel_utils.py:38]: Pretty arbitrary.  I figure 33% headroom felt reasonable. 🤷

Comment (mikelehen) [reply libs/parallel_utils.py:38]: FWIW- I wouldn't be surprised if this parallelization isn't all that productive (especially with maxtasksperchild=1 which I think means we'll be spawning / killing processes a ton) and we could optimize this much better if we did some investigation.  But not worth worrying about for now.  Just want the build to be reliable. 😁

Comment (smcclure17) [reply libs/parallel_utils.py:38]: sounds good! that does make sense

**Pull request merged**